petrometer/petrometer.py

```diff
@@ -219,48 +334,83 @@ def get_db(address: str):
-              f"startblock={start_block}&" \
-              f"endblock=99999999&" \
-              f"page=1&" \
-              f"offset=100&" \
-              f"sort=asc&" \
-              f"apikey={self.arguments.etherscan_api_key}"
+        assert isinstance(address, str)
+        assert isinstance(start_block, int)
+
+        url = (
+            f"https://api.etherscan.io/api?module=account&"
+            f"action=txlist&"
+            f"address={address.lower()}&"
+            f"startblock={start_block}&"
+            f"endblock=99999999&"
+            f"page=1&"
+            f"offset=100&"
+            f"sort=asc&"
+            f"apikey={self.arguments.etherscan_api_key}"
+        )
 
         # Always wait some time before sending a request as we do not want to be banned by etherscan.io
         time.sleep(0.2)
 
         result = requests.get(url, timeout=HTTP_TIMEOUT).json()
 
-        if result['message'] == 'OK':
-            return result['result']
+        if result["message"] == "OK":
+            return result["result"]
 
-        elif result['message'] == 'No transactions found':
+        elif result["message"] == "No transactions found":
             return []
 
         else:
             raise Exception(f"Invalid etherscan.io response: {result}")
 
     @staticmethod
     def get_eth_prices():
-        response = requests.get('https://api.coingecko.com/api/v3/coins/ethereum/market_chart',
-                                params={'vs_currency': 'usd', 'days': 'max', 'interval': 'daily'})
+        response = requests.get(
+            "https://api.coingecko.com/api/v3/coins/ethereum/market_chart",
+            params={"vs_currency": "usd", "days": "max", "interval": "daily"},
+        )
         if response.status_code != 200:
-            raise RuntimeError(f"Failed to fetch historical ETH prices from coingecko.com:"
-                               f" {response.status_code} {response.reason} ({response.text})")
+            raise RuntimeError(
+                f"Failed to fetch historical ETH prices from coingecko.com:"
+                f" {response.status_code} {response.reason} ({response.text})"
+            )
 
-        prices_json = response.json()['prices']
+        prices_json = response.json()["prices"]
         prices = {}
         for price in prices_json:
-            prices[price[0]/1000] = price[1]
+            prices[price[0] / 1000] = price[1]
 
         return prices
 
+    @staticmethod
+    def post_to_grafana(data, graphite_key):
+        # As we can't send test data to Grafana, checking
+        # if test is being ran and avoiding sending test data
+        if "PYTEST_CURRENT_TEST" in os.environ:
+            return True
+
+        # As --graphite-key is not mandatory
+        # checking if it was set
+        if not graphite_key:
+            return True
+
+        result = requests.request(
+            method="POST",
+            url="https://graphite-us-central1.grafana.net/metrics",
```

Comment:
@OleksandrUA Should this be passed as an argument instead? I think the endpoint could be different for different people.